dexpv2/segmentation.py

```diff
@@ -10,6 +11,135 @@
 LOG = logging.getLogger(__name__)
 LOG.setLevel(logging.INFO)
 
+try:
+    import cupy as xp
+
+    LOG.info("cupy found.")
+except (ModuleNotFoundError, ImportError):
+    import numpy as xp
+
+    LOG.info("cupy not found using numpy.")
+
+
+def discretize_multiple_float16_to_uint16(
+    float16_arrays: List["xp.ndarray"],  # type: ignore # 'xp' will be defined at runtime
```

Comment:
I think you could use `ArrayLike` over `xp.ndarray` which corresponds to cupy or numpy arrays for the typing, then you can make your function simpler by not using `globals`, this is also the convention of the library.

Reply:
There's not a lot of benefits of using typing and then `# type: ignore`